WorkloadManagementSystem/Service/JobManagerHandler.py

```diff
@@ -240,6 +246,22 @@ def __deleteJob( self, jobID ):
     if not result['OK']:
       gLogger.warn( 'Failed to delete job from the TaskQueue' )
 
+    # if it was the last job for the pilot, clear PilotsLogging about it
+    result = gPilotAgentsDB.getPilotsForJobID( jobID )
+    if not result['OK']:
+      return result
+    for pilot in result['Value']:
+        res = gPilotAgentsDB.getJobsForPilot( pilot[ 'PilotID' ] )
+        if not res['OK']:
+            return res
+        if not res['Value']: # if list of jobs for pilot is empty, delete pilot and pilotslogging
+            ret = gPilotAgentsDB.deletePilot( pilot[ 'PilotID' ] )
```

Comment:
I think that the pilot logs deletion should be the result of the deletion of jobs, and as a general note, pilots should not be deleted on a time basis (e.g. 10 days after being run).

At the same time, as reminded here, there are pilots that don't match any job, and for which a decision has to be taken. Clearly, we can't leave them forever, so unfortunately for this case I guess we'd need to go back to a time-based deletion.